vehicle/tests/test_main_process.py

```diff
@@ -0,0 +1,61 @@
+import os
+import psutil
+import pickle
+import time
+import zmq
+import yaml
+import multiprocessing as mp
+import threading
+from master_process import MainProcess
+from master_process import _CMD_READ_PATH_KEY, _CMD_READ_KEY_REPLY
+from master_process import _CMD_UPDATE_ROBOT, _CMD_ROBOT_COMMAND
+
+
+def zmq_socket(port=None):
+    context = zmq.Context()
+    socket = context.socket(zmq.REP)
+    if port is None:
+        port = socket.bind_to_random_port("tcp://127.0.0.1")
+    else:
+        socket.bind("tcp://127.0.0.1:{}".format(port))
+    return socket, port
+
+
+def test_main(fixture_path, fixture_robot_command):
```

Comment:
This very preliminary test makes sure the main process can be run and stops gracefully. More can be add later but this is a start point. Note that `pytest` does the magic of making the fixtures available here.